Add integration tile for Mendix

pappachino opened the pull request:
### What does this PR do?

The PR adds a tile for the datadog integration for the Mendix cloud platform.

### Motivation

[Mendix](https://mendix.com/) is a low-code application development platform.
The main purpose is to add visibility for the out-of-the-box Datadog integration for applications deployed to Mendix public cloud.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

eho1307 commented on .github/CODEOWNERS:
```diff
@@ -85,6 +85,7 @@
 /lighthouse/                             @DataDog/agent-integrations
 /logstash/                               @ervansetiawan ervansetiawan@gmail.com
 /logzio/                                 @DataDog/agent-integrations
+/mendix/                                 @mendix/cloud @DataDog/agent-integrations
```
```suggestion
/mendix/                                 @mendix/cloud @DataDog/agent-integrations @DataDog/marketplace-review
```

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+## Overview
+
+[Mendix][1] is a market [leading low-code IDE][2] for professional developers. A powerful visual model-driven development environment to build apps on the Mendix Platform.
```
```suggestion
[Mendix][1] is a [low-code IDE][2] for professional developers. A powerful visual model-driven development environment to build apps on the Mendix Platform.
```

Removing marketing forward copy

pappachino replied:
I've fixed the text

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+With Mendix, you can easily create, change, integrate, test and deploy your applications. Moreover, you can manage branch lines and security as well as extend your app with custom code using the built-in editors.
+
+In addition to the native monitoring capabilities out of the box, Mendix also supports integrations to leading 3rd party application performance management vendors such as DataDog. Refer to our documentation to see all our [supported integrations][3].
```
```suggestion
```

eho1307 replied:
This is not necessary since customers looking at this information already know an integration exists. Please remove this or reword it to 'Refer to [our documentation][3] for more information about monitoring Mendix with APM tools'.

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+In addition to the native monitoring capabilities out of the box, Mendix also supports integrations to leading 3rd party application performance management vendors such as DataDog. Refer to our documentation to see all our [supported integrations][3].
+
+The Mendix integration with DataDog enables you to consume and monitor your Mendix ecosystem. The [DataDog integration][4] from Mendix will enable you to monitor
+Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, SaaS(Software as a Service) environment metrics and any custom metrics you choose to
+configure in Mendix Studio Pro.
```
```suggestion
[This integration][4] allows you to monitor your Mendix ecosystem, including Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, SaaS(Software as a Service) environment metrics. Customize the metrics you send to Datadog in Mendix Studio Pro.
```

eho1307 replied:
Please add in this change for clarity. There's a lot of repetition in the previous version.

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+configure in Mendix Studio Pro.
+
+
+> **Note**:  This integration is applicable to Mendix Cloud & Mendix Cloud Dedicated deployment models.
```
```suggestion
**Note**: This integration is applicable to Mendix Cloud & Mendix Cloud Dedicated deployment models.
```

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+## Setup
+
+To enable the DataDog integration for Mendix applications running on Mendix cloud follow the instructions [here][6].
```
```suggestion
To enable the Datadog integration for Mendix applications running on Mendix cloud follow the instructions [here][6].
```

eho1307 replied:
Please address this comment. Datadog should be spelt with only one uppercase D rather than two.

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+## Data Collected
+
+The list of metrics available for monitoring once the integration is enabled is detailed in the [Mendix documentation][7].
```
If you're sending in metrics, you should include the metadata.csv file that lists the metrics being sent in.

For more info, refer [here](https://docs.datadoghq.com/developers/integrations/check_references/#metrics-metadata-file).

eho1307 replied:
@pappachino Unfortunately, there isn't a way to export a list of metrics with their descriptions.

eho1307 replied:
If you include a list of metrics, we display it on our documentation site so customers can see a list of them on the same page and know what to expect. See an example under the 'Data Collected' section [here](https://docs.datadoghq.com/integrations/1e/).

pappachino replied:
Thanks, @eho1307. I've dropped a metadata.csv file with all the generic metrics available from the Mendix integration.
We won't be able to list all of them, as the metric availability might vary from Mendix app to app.

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+[1]: https://mendix.com/
+[2]: https://www.mendix.com/blog/a-low-code-leader-composing-the-modern-enterprise-with-mendix/
+[3]: https://docs.mendix.com/developerportal/operate/monitoring-with-apm/
```
```suggestion
```

Sakshi-Dhakad replied:
@EH - Is there a reason to remove this link? We would prefer to keep it as it provides an overview of how the integration works with Mendix overall

Sakshi-Dhakad replied:
cc - @pappachino

pappachino replied:
@eho1307 We would like to keep the referenced link as it provides a generic starting point for all APMs including Datadog. I've updated the README accordingly.

eho1307 commented on mendix/manifest.json:
```diff
@@ -0,0 +1,58 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Market leading low-code IDE for application development",
```
```suggestion
    "description": "Monitor Mendix environment metrics",
```
This should be a description of the integration rather than Mendix.

eho1307 replied:
Please remove marketing forward language

eho1307 commented on mendix/manifest.json:
```diff
@@ -0,0 +1,58 @@
+    }   
+    ],
+    "classifier_tags": [
+      "Supported OS::Windows",
+      "Category::Cloud"
```
```suggestion
      "Supported OS::Windows",
      "Category::Cloud",
      "Category::Automation",
      "Offering::Integration"
```

eho1307 replied:
Please address this comment

eho1307 commented on mendix/manifest.json:
```diff
@@ -0,0 +1,58 @@
+      "Category::Cloud"
+    ]
+  },
+  "assets": {},
```
We require integrations to include an out-of-the-box dashboard. Please create one in your sandbox account and export it as a JSON file. The file can be attached to this PR. This ensures customers don't need to set up their own dashboards and receive immediate value after installing the integration. Refer [here](https://datadoghq.dev/integrations-core/guidelines/dashboards/#best-practices) for more info. 

Note: Include a screenshot of your dashboard as an image

pappachino replied:
@eho1307 Thanks for the pointer. 
But currently, we don't have a starter template the customers could use. We would need to give it some thought before we can make it available.

So for the first pass could we publish the integration without any out-of-the-box dashboard ?

pappachino replied:
I've added a dashboard and a sample image.

eho1307 commented on mendix/manifest.json:
```diff
@@ -0,0 +1,58 @@
+    "description": "Market leading low-code IDE for application development",
+    "title": "Mendix",
+    "media": [
+    {
+      "media_type": "image",
+      "caption": "Mendix Studio Pro 9",
+      "image_url": "images/mx9_splash_screen.png"
+    },
```
```suggestion
```

Please only include images that are relevant to the integration.

pappachino replied:
As we don't have any dashboards yet, I guess this is fine for the first pass. We could revisit this once we have screenshots of relevant DD dashboards

eho1307 replied:
Please remove this image

pappachino replied:
I've removed the image.

eho1307 commented on mendix/manifest.json:
```diff
@@ -0,0 +1,58 @@
+      "caption": "Domain model with Data Hub",
+      "image_url": "images/domain-model-with-data-hub.png"
+    },
+    {
+      "media_type": "image",
+      "caption": "Cloud developer portal view of available plans",
+      "image_url": "images/cp_environments_popup.png"
+    },
```
```suggestion
```

Please only include images that are relevant to the integration.

pappachino replied:
As we don't have any dashboards yet, I guess this is fine for the first pass. We could revisit this once we have screenshots of relevant DD dashboards

eho1307 replied:
Please remove this image

eho1307 reviewed (changes requested):
Thank you for opening this PR! I've added comments inline. Feel free to reach out with any questions!

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+
+## Overview
+
+[Mendix][1] is a [low-code IDE][2] for professional developers. A powerful visual model-driven development environment to build apps on the Mendix Platform.
```
```suggestion
[Mendix][1] is a [low-code IDE][2] and visual model-driven development environment that developers can use to build apps on the Mendix Platform.
```

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+
+[Mendix][1] is a [low-code IDE][2] for professional developers. A powerful visual model-driven development environment to build apps on the Mendix Platform.
+
+With Mendix, you can easily create, change, integrate, test and deploy your applications. Moreover, you can manage branch lines and security as well as extend your app with custom code using the built-in editors.
```
```suggestion
With Mendix, you can create, change, integrate, test, and deploy your applications. You can manage branch lines and security as well as extend your app with custom code by using the built-in editor.
```

pappachino replied:
Thanks, I've fixed it.

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+
+With Mendix, you can easily create, change, integrate, test and deploy your applications. Moreover, you can manage branch lines and security as well as extend your app with custom code using the built-in editors.
+
+[This integration][3] allows you to monitor your Mendix ecosystem, including Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, SaaS(Software as a Service) environment metrics. Customize the metrics you send to Datadog in Mendix Studio Pro.
```
```suggestion
[The Mendix integration][3] allows you to monitor your Mendix ecosystem, including Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, and SaaS (Software as a Service) environment metrics. Customize the metrics you send to Datadog in Mendix Studio Pro.
```

pappachino replied:
Thanks, I've fixed it.

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+
+[This integration][3] allows you to monitor your Mendix ecosystem, including Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, SaaS(Software as a Service) environment metrics. Customize the metrics you send to Datadog in Mendix Studio Pro.
+
+**Note**: This integration is applicable to Mendix Cloud & Mendix Cloud Dedicated deployment models.
```
```suggestion
**Note**: This integration is applicable to Mendix Cloud and Mendix Cloud Dedicated deployment models.
```

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+
+## Setup
+
+To enable the Datadog integration for Mendix applications running on Mendix cloud follow the instructions [here][5].
```
```suggestion
To enable the Datadog integration for Mendix applications running on Mendix Cloud, see the [Mendix setup documentation][5].
```

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+
+## Data Collected
+
+The list of metrics available for monitoring once the integration is enabled is detailed in the [Mendix documentation][6].
+
```
```suggestion

### Metrics

For a list of metrics available once the integration is enabled, see the [official Mendix documentation][6].

### Events

The Mendix integration does not include any events.

### Service Checks

The Mendix integration does not include any service checks.

```

pappachino replied:
Thanks, I've fixed it

alai97 commented on mendix/README.md:
```diff
@@ -0,0 +1,33 @@
+The list of metrics available for monitoring once the integration is enabled is detailed in the [Mendix documentation][6].
+
+
+## Support
```
```suggestion
## Troubleshooting
```

pappachino replied:
Fixed

alai97 reviewed (approved):
Some copy suggestions—looks good pending @eho1307's feedback!

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,39 @@
+
+## Overview
+
+[Mendix][1] is a market [leading low-code IDE][2] for professional developers. A powerful visual model-driven development environment to build apps on the Mendix Platform.
```
```suggestion
[Mendix][1] is a [low-code IDE][2] for professional developers. A powerful visual model-driven development environment to build apps on the Mendix Platform.
```

eho1307 replied:
Please remove marketing forward language.

eho1307 replied:
Apologies, ignore this comment. I was looking at an old version of the PR.

eho1307 commented on mendix/CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+# CHANGELOG - Mendix
+
+## 1.0.0
+* [Added] Initial revision of Mendix integration
```
Please address this comment

pappachino commented:
@eho1307 @alai97  I've addressed most of the comments as suggested. Do let me know if you have any further comments and if the PR can be accepted. Thanks for reviewing the PR

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,47 @@
+
+## Overview
+
+[Mendix][1] is a [low-code IDE][2] and a visual model-driven development
+environment that developers can use to build apps on the Mendix Platform.
```
```suggestion
[Mendix][1] is a [low-code IDE][2] and a visual model-driven development environment that developers can use to build apps on the Mendix Platform.
```

eho1307 commented on mendix/README.md:
```diff
@@ -0,0 +1,47 @@
+[Mendix][1] is a [low-code IDE][2] and a visual model-driven development
+environment that developers can use to build apps on the Mendix Platform.
+
+With Mendix, you can easily create, change, integrate, test, and deploy your
+applications. You can manage branch lines and security as well as extend your
+app with custom code by using the built-in editors.
```
```suggestion
With Mendix, you can easily create, change, integrate, test, and deploy your applications. You can manage branch lines and security as well as extend your app with custom code by using the built-in editors.
```

emarsha94 commented on mendix/README.md:
```diff
@@ -0,0 +1,43 @@
+
+## Overview
+
+[Mendix][1] is a [low-code IDE][2] and a visual model-driven development environment that developers can use to build apps on the Mendix Platform.
+
+With Mendix, you can easily create, change, integrate, test, and deploy your applications. You can manage branch lines and security as well as extend your app with custom code by using the built-in editors.
+
+[The Mendix integration][3] allows you to monitor your Mendix ecosystem, including Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, SaaS(Software as a Service) environment metrics. Customize the metrics you send to Datadog in Mendix Studio Pro.
```
```suggestion
[Mendix][1] is a [low-code IDE][2] and a visual model-driven development environment that developers can use to build apps on the Mendix Platform. With Mendix, you can easily create, change, integrate, test, and deploy your applications. You can manage branch lines and security as well as extend your app with custom code by using the built-in editors.

[The Mendix integration][3] allows you to monitor your Mendix ecosystem, including Mendix Runtime metrics, Java Virtual Machine (JVM) metrics, database, and SaaS(Software as a Service) environment metrics. You can customize the metrics you send to Datadog in Mendix Studio Pro.
```

emarsha94 replied:
Can you provide insight on the value of monitoring the metrics?

emarsha94 reviewed (changes requested):
Could you provide your logo in svg format? The light mode logo is cut off. Thanks!

emarsha94 commented:
> Hi @pappachino The only remaining comment from my end is regarding the dashboard. Once that's been added in, your PR is good to go from my end!

Hi @pappachino - Could you add a dashboard to the integration? Thanks!

pappachino commented:
> > Hi @pappachino The only remaining comment from my end is regarding the dashboard. Once that's been added in, your PR is good to go from my end!
> 
> Hi @pappachino - Could you add a dashboard to the integration? Thanks!

@emarsha94  we'll be adding a dashboard to the PR next week. Also, I'll provide svg format logo. 
Thanks for the review.

pappachino commented:
> Could you provide your logo in svg format? The light mode logo is cut off. Thanks!

The logo was already in svg format. Can you check again. Thanks!

pappachino commented:
> Hi @pappachino The only remaining comment from my end is regarding the dashboard. Once that's been added in, your PR is good to go from my end!

I've added a dashboard. Let me know if there is anything else, thanks!

pappachino commented:
@emarsha94  I've addressed the review comments. 
Do let me know if you have any further comments and if the PR can be accepted.

emarsha94 commented on mendix/manifest.json:
```diff
@@ -0,0 +1,70 @@
+        "image_url": "images/mendix_dashboard.png"
+    }
+    ],
+    "classifier_tags": [
+      "Supported OS::Windows",
+      "Category::Cloud",
+      "Category::Automation",
+      "Offering::Integration"
```
```suggestion
    "classifier_tags": [
      "Supported OS::Windows",
      "Category::Cloud",
      "Category::Automation",
      "Offering::Integration",
      “Submitted Data Type::Metrics”
```

emarsha94 reviewed (changes requested):
Added 2 more suggestions. Once those are added, this looks good to me :) Thanks!

pappachino commented:
> Added 2 more suggestions. Once those are added, this looks good to me :) Thanks!

I've addressed the suggestions.

emarsha94 commented on mendix/README.md:
```diff
@@ -0,0 +1,43 @@
+
+**Note**: This integration is applicable to Mendix Cloud and Mendix Cloud Dedicated deployment models.
+
+
```
```suggestion
```

emarsha94 commented on mendix/CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+# CHANGELOG - Mendix
+
+## 1.0.0 / 2023-03-27
```
```suggestion
## 1.0.0 / 2023-03-31
```

emarsha94 reviewed (approved):
Looks good to me! Two minor changes requested. Thanks!
